src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -2296,105 +2296,117 @@ private void handleStateNormal(final InetAddress endpoint, final String status)
         // Order Matters, TM.updateHostID() should be called before TM.updateNormalToken(), (see CASSANDRA-4300).
         UUID hostId = Gossiper.instance.getHostId(endpoint);
         InetAddress existing = tokenMetadata.getEndpointForHostId(hostId);
-        if (replacing && isReplacingSameAddress() && Gossiper.instance.getEndpointStateForEndpoint(DatabaseDescriptor.getReplaceAddress()) != null
-            && (hostId.equals(Gossiper.instance.getHostId(DatabaseDescriptor.getReplaceAddress()))))
-            logger.warn("Not updating token metadata for {} because I am replacing it", endpoint);
-        else
-        {
-            if (existing != null && !existing.equals(endpoint))
+        boolean acquiredTokenLock = false;
```

Comment:
This diff is hard to read.  Did anything significantly change besides adding the lock?

Reply:
Nope, only the lock, unlock, and the `try {} finally {}`